### PR TITLE
Add magic-wormhole

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
pending https://github.com/conda-forge/python-spake2-feedstock/pull/3

File: recipes/magic-wormhole/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    - pip
+
+
+test:
+  commands:
+    - wormhole --version
+
+
+about:
+  home: http://github.com/simplejson/simplejson
```
Comment (ericdill) [inline recipes/magic-wormhole/meta.yaml]: wrong url

Comment (CJ-Wright) [inline recipes/magic-wormhole/meta.yaml]: I really need to write a piece of the linter for catching things from the examples.

**Comment by ericdill** [comment]
So you could add a patch to setup.py that would delete the spake2 requirement since you're already capturing that in your meta.yaml but then you'll hit another error of the same variety

```
+ wormhole --version
Traceback (most recent call last):
  File "/Users/edill/miniconda/conda-bld/magic-wormhole_1495565696253/_t_env/bin/wormhole", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/edill/miniconda/conda-bld/magic-wormhole_1495565696253/_t_env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3019, in <module>
    @_call_aside
  File "/Users/edill/miniconda/conda-bld/magic-wormhole_1495565696253/_t_env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3003, in _call_aside
    f(*args, **kwargs)
  File "/Users/edill/miniconda/conda-bld/magic-wormhole_1495565696253/_t_env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3032, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/edill/miniconda/conda-bld/magic-wormhole_1495565696253/_t_env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 655, in _build_master
    ws.require(__requires__)
  File "/Users/edill/miniconda/conda-bld/magic-wormhole_1495565696253/_t_env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 963, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/edill/miniconda/conda-bld/magic-wormhole_1495565696253/_t_env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 849, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'idna>=0.6' distribution was not found and is required by twisted
TESTS FAILED: magic-wormhole-0.9.1-py36_0
```

I'm not even sure how twisted is getting pulled in 🤷‍♂️ 

setup.py.patch:
```diff
diff --git a/setup.py b/setup.py
index 574aa4a..ccd2ac5 100644
--- a/setup.py
+++ b/setup.py
@@ -27,7 +27,7 @@ setup(name="magic-wormhole",
           ]
       },
       install_requires=[
-          "spake2==0.7", "pynacl",
+          "pynacl",
           "six",
           "twisted[tls]",
           "autobahn[twisted] >= 0.14.1",
```

and then add this to the source section:

```yaml
  patches:
    - setup.py.patch
```

I guess keep going down the rabbit hole of patching the setup.py 🤷‍♂️

**Comment by CJ-Wright** [comment]
Can I just remove the entire `install_requires` section?

**Comment by CJ-Wright** [comment]
Pending #3203

**Comment by ericdill** [comment]
@CJ-Wright was this still of interest?

**Comment by CJ-Wright** [comment]
Interest yes, but I don't know if I'll be able to work on it in the near future. You are more than welcome to take this over it close it if you want.

**Pull request closed without merging**